surfinBH/_fit_evaluators/fit_3dq8BMS.py

```diff
@@ -0,0 +1,236 @@
+        return self._eval_wrapper('chif', *args, **kwargs)
+    
+    def vf(self, *args, **kwargs):
+        """vf is not implemented in this model. Will return (None, None),"""
+        return self._eval_wrapper('vf', *args, **kwargs)
+
+    def all(self, *args, **kwargs):
+        """ Evaluates fit and 1-sigma error estimate for supertranslation parameter alpha, and boost velocity 
+        of the BMS transformation from the inspiral BMS frame to the remnant BMS frame
+        Returns:
+            alpha, boost_vel,  alpha_err, boost_vel_err
+        """
+        return self._eval_wrapper('all', *args, **kwargs)
+
+    #-------------------------------------------------------------------------
+    def _eval_wrapper(self, fit_key, q, chiA, chiB, **kwargs):
+        """ Evaluates the NRSur3dq8BMSRemnant model.
+        """
+        chiA = np.array(chiA)
+        chiB = np.array(chiB)
+
+        # Warn/Exit if extrapolating
+        allow_extrap = kwargs.pop('allow_extrap', False)
+        self._check_param_limits(q, chiA, chiB, allow_extrap)
+
+        self._check_unused_kwargs(kwargs)
+
+        x = [q, chiA[2], chiB[2]]
+
+        ell_max = 8
+        sur_predictions_positive_m_modes = {}
+        sur_predictions_positive_m_modes_err = {}
+        positive_m_modes_indices = get_non_negative_modes_indices(ell_min = 2,ell_max = ell_max)
+        zero_m_modes_indices = get_zero_modes_indices(ell_min = 2, ell_max = ell_max)
+        for i in positive_m_modes_indices:
+            sup_real, sup_real_err = self._evaluate_fits(x, f'sup_real({i})')
+            if i not in zero_m_modes_indices:
+                sup_imag, sup_imag_err = self._evaluate_fits(x, f'sup_imag({i})')
+                sur_predictions_positive_m_modes[i] = sup_real + (1j)*sup_imag
+                sur_predictions_positive_m_modes_err[i] = sup_real_err + (1j)*sup_imag_err
+            else:
+                sur_predictions_positive_m_modes[i] = sup_real  
+                sur_predictions_positive_m_modes_err[i] = sup_real_err 
+        alpha = get_full_supertranslation_array(sur_predictions_positive_m_modes,ell_max)
+        alpha_err = get_full_supertranslation_array(sur_predictions_positive_m_modes_err,ell_max)
+
+        boost_velx, boost_velx_err = self._evaluate_fits(x, 'boost_velx')
+        boost_vely, boost_vely_err = self._evaluate_fits(x, 'boost_vely')
+        boost_velz, boost_velz_err = self._evaluate_fits(x, 'boost_velz')
+        boost_vel = np.array([boost_velx, boost_vely, boost_velz])
+        boost_vel_err = np.array([boost_velx_err, boost_vely_err, boost_velz_err])
+
+        if fit_key == 'mf':
+            mf, mf_err = None, None
+            return mf, mf_err
+        
+        if fit_key == 'chif':
+            chif, chif_err = None, None
+            return chif, chif_err
+        
+        if fit_key == 'vf':
+            vf, vf_err = None, None
+            return vf, vf_err
```

Comment:
Handle these cases first to avoid doing computations if the user calls mf, chif, or vf?